validation/src/verifier/event_verifier.rs

```diff
@@ -287,4 +320,174 @@ mod test {
+
+        let mut buffer = Vec::new();
+        let mut writer = CarWriter::new(header, &mut buffer);
+        writer.write(envelope_cid, envelope).unwrap();
+        writer.write(payload_cid, payload).unwrap();
+        writer.write(cacao_cid, cacao).unwrap();
+        writer.finish().unwrap();
+        let content = multibase::encode(multibase::Base::Base36Lower, buffer);
+        panic!("{}", content);
+
+        */
+        let (_, buffer) = multibase::decode(GITCOIN_INIT_EVENT_CAR).unwrap();
+        let (_cid, event) =
+            unvalidated::Event::<Ipld>::decode_car(std::io::Cursor::new(buffer), false).unwrap();
+
+        match event {
+            unvalidated::Event::Time(_) => unreachable!(),
+            unvalidated::Event::Signed(event) => {
+                let controller = match event.payload() {
+                    unvalidated::Payload::Data(_) => unreachable!(),
+                    unvalidated::Payload::Init(payload) => {
+                        payload.header().controllers().first().unwrap()
+                    }
+                };
+                assert_eq!(
+                    controller,
+                    "did:pkh:eip155:1:0xc1e722551c7eac8675903f95f1330a2fe6ad34ea"
+                );
+                event
+                    .verify_signature(Some(controller), &VerifyJwsOpts::default())
+                    .await
+                    .expect("event should be valid");
+            }
+            unvalidated::Event::Unsigned(_) => unreachable!(),
+        }
+    }
+
+    #[tokio::test]
+    /// The event used to verify in js-did. Demonstrates how you can parse the dag-json envelope and cacao to verify the signatureca
+    async fn gitcoin_cacao_from_pieces() {
+        // dag-json event
+        let event = serde_json::json!({
+            "link": {
+                "/": "bafyreiezmymvqjtss5emo4uuj7emmoxnhpm3kjukzodahp5rqij5xsreo4"
+            },
+            "payload": "AXESIJlmGVgmcpdIx3KUT8jGOu072bUmisuGA7-xghPbyiR3",
+            "signatures": [
+                {
+                "protected": "eyJhbGciOiJFUzI1NiIsImNhcCI6ImlwZnM6Ly9iYWZ5cmVpYWdvNWQ2NzRsaWxsc2piZHp2bXNidDNva3dhcmdqZ2VrY2V2c3U3d3h1endmY3h6NDd4YSIsImtpZCI6ImRpZDprZXk6ekRuYWVVM3o2d0pLWWN4ZGVRZE5CMWI2YmduVVpCdE55NnU4VzdBVDNyVlBkd2oyZiN6RG5hZVUzejZ3SktZY3hkZVFkTkIxYjZiZ25VWkJ0Tnk2dThXN0FUM3JWUGR3ajJmIn0",
+                "signature": "F4V-EpyUujZGIrgx0bify8AWMgAVviNFMB5vHW7oxEk0OyFsl4UKmmzLKb2bN0GnUJi0rrkUy8FmRFqgPgEWQQ"
+                }
+            ]
+        });
+        let event = event.as_object().unwrap();
+        // bafyreiago5d674lillsjbdzvmsbt3okwargjgekcevsu7wxuzwfcxz47xa from the protected data
+        let cacao = r#"{
+            "h": {
+              "t": "eip4361"
+            },
+            "p": {
+              "aud": "did:key:zDnaeU3z6wJKYcxdeQdNB1b6bgnUZBtNy6u8W7AT3rVPdwj2f",
+              "domain": "app.passport.xyz",
+              "exp": "2024-12-24T20:27:13.330Z",
```

Comment:
TODO: update these tests to use a time check before the exp date so these don't start failing next week